src/Compilers/CSharp/Test/Syntax/Parsing/ParsingTests.cs

```diff
@@ -15,14 +15,35 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
 {
     public abstract class ParsingTests : CSharpTestBase
     {
-        private IEnumerator<SyntaxNodeOrToken> _treeEnumerator;
+        private IEnumerator<SyntaxNodeOrToken>? _treeEnumerator;
         private readonly ITestOutputHelper _output;
+        private bool _dump = false;
```

Comment:
Since we're in here..
Why do we need to set this flag in the first place? Couldn't we dump the expected result upon failure, as we do for other tests?

Reply:
The way it's factored now, Print is "baked in" to the enumeration of the tree. And that does simplify some things, especially printing curly braces in the appropriate spots, but it means there's no decent way (that I know) to "communicate" with the enumerator without setting a flag.

Reply:
Thinking a bit more, I think I realize what you're suggesting--perhaps we could add a 'dump' parameter to EnumerateNodes, and in the event of a failure (unexpected node, unconsumed node), just "restart" the tree and dump the whole thing so you can paste it in.